docs/design/2022-03-10-write-bypass-tidb.md

```diff
@@ -0,0 +1,117 @@
+# Flink connector write bypass TiDB
+
+- Author(s): [yangxin](http://github.com/xuanyu66)
+- Tracking Issue: https://github.com/tidb-incubator/TiBigData/issues/163
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Motivation or Background](#motivation-or-background)
+* [Detailed Design](#detailed-design)
+* [Compatibility](#compatibility)
+* [Test Design](#test-design)
+    * [Functional Tests](#functional-tests)
+    * [Scenario Tests](#scenario-tests)
+    * [Compatibility Tests](#compatibility-tests)
+    * [Benchmark Tests](#benchmark-tests)
+* [Impacts & Risks](#impacts--risks)
+* [Investigation & Alternatives](#investigation--alternatives)
+* [References](#references)
+
+## Introduction
+
+This article introduces an optimized solution for writing Flink connector to TiDB cluster, by bypassing TiDB server and writing TiKV directly.
+
+## Motivation or Background
+
+Currently, Fink connector uses JdbcDynamicTableSink to write data through TiDB server.
+
+However, this approach leads to the following problems.
+- Lack of atomicity, if the Flink task fails and exists during the writing process, some data will be written successfully while the others not.
+- Lack of isolation, other transactions can see part of the written data during the writing process.
+- Lack of failure recovery mechanism, tasks need to be idempotency, for example, after failure, you need to clean up written data first, and then re-run the Flink program.
+- Latency is higher than directly writing to TiKV.
+- Streaming or batch processing of large data will affect the performance of TiDB server and thus affect online business queries.
+
+In contrast to this, Fink connector direct-write TiKV can solve these problems.
+- TiKV provides transaction support for atomic, isolation, and failure recovery.
+- Writing directly to TiKV has lower latency and thus does not affect the TiDB server.
+
+In terms of feasibility, Spark connector has proven the feasibility of writing bypass TiDB.
+
+## Detailed Design
+
+### Add TiDBDynamicTableSink option
+
+In the factory method of TiDBDynamicTableFactory, we will add TiDBDynamicTableSink option. The user can choose how to write data by setting the Sink mode (JDBC or TiKV). 
+
+### Implement different dimensional transaction levels
+
+TiKV implements Google's Percolator transaction mode, which is a variation of 2pc. See [tidb-transaction-model](https://pingcap.com/zh/blog/tidb-transaction-model). 
```

Comment:
Maybe we could use an English description, like https://tikv.org/deep-dive/distributed-transaction/introduction/